datacube_stats/utils/__init__.py

```diff
@@ -96,6 +96,21 @@ def to_float(da):
     return data.apply(to_float, keep_attrs=True)
 
 
+def restore_nodata(var_name, var, measurements):
```

Comment:
I would prefer a smart cast function that took care of `NaN` and returned `dataarray` with proper `dtype`

Here we are assuming that `nodata` value is properly representable in `float32`, it might not be, not all `int32` values can be stored in `float32` variable for example.

Reply:
Currently the task of conversion is performed by `datacube_stats.output_drivers.NetCDFCFOutputDriver` anyway. The smart cast function would have to duplicate that functionality outside the drivers.

Reply:
Done implementing the smart cast. Requesting review.